doc/mgmt/Management Framework.md

```diff
@@ -2378,54 +2469,68 @@ mechanism for the app specific configuration.
+39. REST response is sent by the REST gateway to the REST client.
 
 ### 4.2 REST GET flow
 
-![REST GET flow](images/read.jpg)
+![REST GET flow](images/read_mdb.jpg)
 
 1. REST GET request from the REST client is sent to the REST Gateway.
 2. REST Gateway invokes a common request handler.
 3. Authentication of the incoming request is performed.
 4. Request handler calls the Translib exposed GET API with the uri of the request.
 5. Translib infra gets the App module corresponding to the incoming uri.
-6. Translib infra calls the initialize function of the App module with the YGOT structures and path. App module caches them.
-7. Status retuned from App module.
-8. App module queries Transformer to translate the path to the Redis keys that need to be queried.
-9. Status returned from App module.
-10. Translib infra calls the processGet function on the App module
-11. App modules calls read APIs exposed by the DB access layer to read data from the Redis DB.
-12. Data is read from the Redis DB is returned to the App module
-13. App module fills the YGOT structure with the data from the Redis DB and validated the filled YGOT structure for the syntax.
-14. App module converts the YGOT structures to JSON format.
-15. IETF JSON payload is returned to the Translib infra.
-16. IETF JSON payload is returned to the request handler.
-17. Response is returned to REST gateway.
-18. REST response is returned to the REST client from the REST gateway.
+6. Translib infra calls the getNamespace function of the App module with the uri path, to get all DB namespace associated with the uri path
+7. App module queries Transformer to get all namespace associated with the uri path
+8. Transformer shall use Yang specific "Transformer Spec" to get all namespace associated with uri path
+9. Transformer return namespace associated with the uri path
+10. App module return namespace retrived by the transformer layer
+11. Translib infra gets all the namespaces available in the platform, loops through each of the namespace associated with the uri path as recieved in earlier step
+12. Translib infra calls the initialize function of the App module with the YGOT structures and path. App module caches them.
+13. Status retuned from App module.
+14. App module queries Transformer to translate the path to the Redis keys that need to be queried.
+15. Status returned from App module.
+16. Translib infra calls the processGet function on the App module
+17. App modules calls read APIs exposed by the DB access layer to read data from the Redis DB.
+18. Data is read from the Redis DB is returned to the App module
+19. App module fills the YGOT structure with the data from the Redis DB and validated the filled YGOT structure for the syntax.
+20. App module converts the YGOT structures to JSON format.
+21. IETF JSON payload is returned to the Translib infra.
+22. Translib infra aggregates all the responses from app module with different namesapce DB associated with uri path
```

Comment:
The above explanation is generic handling, in case of path with specific keys the namespace will return unique namespace only and hence it will be single response.

> This breaks the RESTCONF and gNMI specs. Cannot aggregate the values data if the path points to a specific leaf.
> 
> For example -- what will be the ourtput of GET /interfaces/interface[name=Ethernet0]/state/mtu if this path was mapped to multiple namespaces? Client is supposed to receive only one value.